components/region_cache_memory_engine/src/background.rs

```diff
@@ -451,6 +603,76 @@ impl BackgroundRunnerCore {
         #[cfg(test)]
         flush_epoch();
     }
+
+    fn evict_on_soft_limit_reached(&mut self) {
```

Comment:
TODO: document. Again, exact test here may be difficult.. will need to use mocks.